org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormatterSettings.java

```diff
@@ -0,0 +1,131 @@
+package org.eclipse.lsp4xml.settings;
+
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+import javax.swing.text.TabableView;
+
+import org.eclipse.lsp4j.FormattingOptions;
+
+/**
+ * This class is the root of all formatting settings.
+ * It is necessary to update this class for any new additions.
+ * 
+ * All defaults should be set here to eventually be overriden if needed.
+ */
+public class XMLFormatterSettings {
+  private static final Logger LOGGER = Logger.getLogger(XMLFormatterSettings.class.getName());
+
+  //All possible keys
+  public final static String TAB_SIZE = "tabSize";
+  public final static String INSERT_SPACES = "insertSpaces";
+  public final static String SPLIT_ATTRIBUTES = "splitAttributes";
+  public final static String JOIN_CDATA_LINES = "joinCDATALines";
+  public final static String FORMAT_COMMENTS = "formatComments";
+  public final static String JOIN_COMMENT_LINES = "joinCommentLines";
+
+  private static int defaultTabSize = 4;
+  private static boolean defaultIndentSpaces = false;
+  private Map<String, Object> settingsMap;
```

Comment:
Remove this map if you extends from FormattingOptions which extends already of HashMap